flutter_app/lib/widgets/nav_bar_widget.dart

```diff
@@ -0,0 +1,64 @@
+import 'package:flutter/material.dart';
+
+class NavBar extends StatefulWidget {
+  const NavBar({super.key});
+
+  @override
+  NavBarState createState() => NavBarState();
+}
+
+class NavBarState extends State<NavBar> {
+  int _index = 0;
```

Comment:
If your going with this implementation, I would actually recommend getting rid of the router and replacing it with a list or map of screen widgets. You'd essentially be replacing the router and making a single screen app with interchangeable screens. Since we are using a router there is some weirdness with the index.